src/table.ts

```diff
@@ -1863,6 +1872,95 @@ class Table extends common.ServiceObject {
         ? optionsOrCallback
         : (cb as InsertRowsCallback);
 
+    this._insertWithRetry(rows, options)
+      .catch(err => {
+        if ((err as common.ApiError).code === 404 && options.schema) {
+          return this._createTableAndInsert(rows, options);
+        }
+        throw err;
+      })
+      .then(
+        resp => callback(null, resp),
+        err => callback(err, null)
+      );
+  }
+
+  /**
+   * Attempts to create the table before inserting the rows.
+   *
+   * @private
+   *
+   * @param {object|object[]} rows The rows to insert.
+   * @param {object} options Insert options.
+   * @returns {Promise}
+   */
+  private async _createTableAndInsert(
+    rows: RowMetadata | RowMetadata[],
+    options: InsertRowsOptions
+  ) {
+    const {schema} = options;
+    const delay = 60000;
+
+    try {
+      await this.create({schema});
+    } catch (e) {
+      if (e.code !== 409) {
+        throw e;
+      }
+    }
+
+    await new Promise(resolve => setTimeout(resolve, delay));
+    return (await this.insert(rows, options))[0];
```

Comment:
Could we make this a little more readable?  This could be 2-3 lines of code that more sense :)

Reply:
refactored as requested.